utils.go

```diff
@@ -666,3 +666,20 @@ func (h *hashReaderWrapper) Read(p []byte) (n int, err error) {
 	}
 	return n, err
 }
+
+// wrapReader embeds io.Reader but returns an error
+// with 'read error' prefix. This is needed because
+// net/http Do() request does not have a way to indicate
+// a way to indicate this is an error from reading the
+// request body.
+type wrapReader struct {
+	io.Reader
+}
+
+func (wr wrapReader) Read(p []byte) (n int, e error) {
+	n, e = wr.Reader.Read(p)
+	if e != nil && e != io.EOF { // do not wrap io.EOF for callers not using errors.Is() yet
```

Comment:
maybe also io.ErrUnexpectedEOF @vadmeste

Reply:
> maybe also io.ErrUnexpectedEOF @vadmeste

I don't worry too much about io.ErrUnexpectedEOF because the caller will have to stop the execution and return the error but it is not the case with with io.EOF